bookwyrm_backend/src/main/java/com/bookwyrm/backend/comment/controller/CommentController.java

```diff
@@ -0,0 +1,44 @@
+package com.bookwyrm.backend.comment.controller;
+
+import com.bookwyrm.backend.comment.input.CommentUploadInput;
+import com.bookwyrm.backend.comment.payload.CommentUploadPayload;
+import com.bookwyrm.backend.comment.validator.CommentUploadValidator;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/comment")
+public class CommentController {
+    /*
+     * Requires some tests:
+     * Happy case (everything works as expected)
+     * Missing Author
+     * Missing Comment Description
+     * Missing Comment Anonymous Flag
+     * Missing Everything
+     * */
+    @CrossOrigin
+    @PostMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<CommentUploadPayload> createComment(
+            @RequestBody CommentUploadInput commentUploadInput,
+            HttpServletRequest request) {
+        CommentUploadPayload response = new CommentUploadPayload();
+        List<String> errorList = CommentUploadValidator.validateUploadInformation(commentUploadInput);
+        HttpStatus status = HttpStatus.OK;
+
+        if (errorList.isEmpty()) {
+            response.setCommentAuthor(commentUploadInput.getAuthor());
+            response.setCommentDescription(commentUploadInput.getDescription());
+            response.setCommentAnonymousFlag(commentUploadInput.getAnonymousFlag());
```

Comment:
Eventually we'll add a view item for comments that can be auto populated by giving it the dao (of business object). Though that'll probably wait until the backend change branch.